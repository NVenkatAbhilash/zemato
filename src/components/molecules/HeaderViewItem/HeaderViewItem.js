import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import { Grid } from "@material-ui/core";
import Typography from "../../atoms/Typography/Typography";

const useStyles = makeStyles({
  root: {
    border: "1px solid rgb(0,0,0,.1)",
    borderRadius: "15px",
    maxWidth: "500px",
    cursor: "pointer",
    transition: "transform .3s",
    "&:hover": {
      transform: "scale(1.08)",
    },
  },
  image: {
    objectFit: "cover",
    width: "100%",
    height: "300px",
    borderRadius: "15px 15px 0 0",
  },
  content: {
    padding: "8px 0 15px 0",
    color: "rgb(0,0,0,.8)",
  },
});

const HeaderViewItem = ({ foodImg, children, ...props }) => {
  const classes = useStyles();

  //const foodImg = require('./foodImg.png');

  return (
    <Grid
      container
      direction="column"
      alignItems="center"
      className={classes.root}
      {...props}
    >
      <Grid item>
        <img alt="view type" className={classes.image} src={foodImg} />
      </Grid>
      <Grid item className={classes.content}>
        <Typography variant="h6" data-testid="children" data-testid="text">
          {children}
        </Typography>
      </Grid>
    </Grid>
  );
};

export default HeaderViewItem;
