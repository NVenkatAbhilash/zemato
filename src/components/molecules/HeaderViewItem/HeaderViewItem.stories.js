import React from "react";
import HeaderViewItem from "./HeaderViewItem";
import foodImg from "./../../../resources/images/foodItem.jpg";


export default {
  title: "molecules/HeaderViewItem",
  component: HeaderViewItem,
};

const Template = ({ ...args }) => <HeaderViewItem {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  children: "Food Item",
  foodImg: foodImg,
};
