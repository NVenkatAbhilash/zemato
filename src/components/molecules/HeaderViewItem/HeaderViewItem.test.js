import React from "react";
import ReactDOM from "react-dom";
import { render, fireEvent, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import HeaderViewItem from "./HeaderViewItem";
import foodImg from "./../../../resources/images/foodItem.jpg";
import renderer from "react-test-renderer";

it("renders HeaderViewItem snapshot correctly", () => {
  const args = {
    children: "Food Item",
    foodImg: foodImg,
  };

  const tree = renderer.create(<HeaderViewItem {...args} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders HeaderViewItem without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<HeaderViewItem />, div);
});

it("renders HeaderViewItem correctly", () => {
  const args = {
    children: "Food Item",
    foodImg: foodImg,
  };
  const { getByTestId } = render(<HeaderViewItem {...args} />);
  expect(getByTestId("text")).toHaveTextContent(args.children);
});
