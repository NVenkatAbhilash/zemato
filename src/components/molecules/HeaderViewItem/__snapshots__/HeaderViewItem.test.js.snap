// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders HeaderViewItem snapshot correctly 1`] = `
<div
  className="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <img
      alt="view type"
      className="makeStyles-image-2"
      src="foodItem.jpg"
    />
  </div>
  <div
    className="MuiGrid-root makeStyles-content-3 MuiGrid-item"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6"
      data-testid="text"
    >
      Food Item
    </h6>
  </div>
</div>
`;
